### DISCUSSION TITLE
What is the license of this repository?

### DISCUSSION THREAD

**Discussion by Maniues** [dotnet/csharplang#4855] [Q&A]
What is the license of this repository?

**Comment by TheUnlocked** [comment]
This repository is primarily for discussion, but the C# specification has [all rights reserved by Microsoft](https://www.github.com/dotnet/csharplang/tree/main/spec%2FLICENSE.md).

**Reply by Maniues** [reply]
Why it's all right Reserved when ECMA standard is MIT/CCBY licensed and c# docs are CCBY licensed?

**Reply by Joe4evr** [reply]
My guess is because "Microsoft's C#" does *more* than what ECMA-334 specifies, which the team does plan to remedy eventually (see [all these features that still needs ECMA spec](https://github.com/dotnet/csharplang/issues?q=is%3Aopen+is%3Aissue+label%3A%22Implemented+Needs+ECMA+Spec%22)), but they haven't gotten there yet. So all the features that are outside of ECMA are All Rights Reserved until the spec catches up.

**Reply by MadsTorgersen** [reply]
The Microsoft spec and the ECMA standard were historically two separate documents; hence the discrepancy. The intention is to only have the ECMA document going forward, and we plan to remove the (draft) Microsoft version here in csharplang when the ECMA version (which is now [open source](https://github.com/dotnet/csharpstandard)) has caught up.

**Reply by Maniues** [reply]
> The Microsoft spec and the ECMA standard were historically two separate documents; hence the discrepancy. The intention is to only have the ECMA document going forward, and we plan to remove the (draft) Microsoft version here in csharplang when the ECMA version (which is now [open source](https://github.com/dotnet/csharpstandard)) has caught up.

Thanks for the answer from someone from Microsoft.  So I have a question: if this (draft) is to be deleted, why should it remain "all rights reserved"?  Can't you share this on CCBY and the code on MIT and remove csharplang by archiving this repo?  Why should your work (publicly available through the .NET Foundation) be restricted?  I guess the goal of the .NET Foundation is to share software under a free license, right?

**Reply by HaloFour** [reply]
@MadsTorgersen 

Would it be safe to assume that this repo will likely always be a step (or more) ahead of the ECMA standard?  Especially for any proposals still under design or implementation?  Wouldn't those documents fall under a different license?
